pkg/crd/crd.go

```diff
@@ -362,6 +361,74 @@ func (c *Client) UpgradeAll() error {
 			return err
 		}
 	}
+
+	updatedAzureIdentities := make(map[string]aadpodv1.AzureIdentity)
+	updatedAzureIdentityBindings := make(map[string]aadpodv1.AzureIdentityBinding)
+	i, err := c.getObjectList(aadpodv1.AzureIDResource, &aadpodv1.AzureIdentityList{})
+	if err != nil {
+		return err
+	}
+	list, err := meta.ExtractList(i)
+	if err != nil {
+		return fmt.Errorf("extract list error for resource: %s, err: %v", aadpodv1.AzureIDResource, err)
+	}
+	for _, item := range list {
+		o, err := meta.Accessor(item)
+		if err != nil {
+			return fmt.Errorf("get object for resource: %s, error: %v", aadpodv1.AzureIDResource, err)
+		}
+		updatedAzureIdentities[o.GetNamespace()+"/"+o.GetName()] = *o.(*aadpodv1.AzureIdentity)
```

Comment:
Can we populate the cache directly when doing the update of the AzureIdentity and Binding instead of doing another fetch and parse again. Currently the "Into" of the setObject is not used. We can use that fill the cache avoiding the second set of calls.